spec/policy/find_service_spec.lua

```diff
@@ -0,0 +1,108 @@
+local configuration = require('configuration')
+
+describe('find_service', function()
+  describe('.rewrite', function()
+    describe('when path routing is enabled', function()
+      it('finds the service by matching rules and stores it in the given context', function()
+        require('configuration_store').path_routing = true
+
+        -- We access ngx.var.request and ngx.req.get_uri_args directly in the
+        -- code, so we need to mock them. We should probably try to avoid this
+        -- kind of coupling.
+        ngx.var = { request = 'GET /abc HTTP/1.1' }
+        ngx.req = { get_uri_args = function() return {} end }
+
+        local find_service_policy = require('policy.find_service').new()
+        local host = 'example.com'
+
+        local service_1 = configuration.parse_service({
+          id = 42,
+          proxy = {
+            hosts = { host },
+            proxy_rules = { {
+              pattern = '/abc',
+              http_method = 'GET',
+              metric_system_name = 'hits',
+              delta = 1
+            } }
+          }
+        })
+
+        local service_2 = configuration.parse_service({
+          id = 43,
+          proxy = {
+            hosts = { host },
+            proxy_rules = { {
+              pattern = '/def',
+              http_method = 'GET',
+              metric_system_name = 'hits',
+              delta = 2
+            } }
+          }
+        })
+
+        local configuration_store = require('configuration_store').new()
+        configuration_store:store({ services = { service_1, service_2 } })
```

Comment:
I think it would be better to test with 3 services. Otherwise, we won't know if this works by coincidence because it just returns the first or the last inserted.

Reply:
Done.